pkg/geo/geogfn/segmentize.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geogfn
+
+import (
+	"math"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/cockroach/pkg/geo/geographiclib"
+	"github.com/cockroachdb/errors"
+	"github.com/golang/geo/s2"
+	"github.com/twpayne/go-geom"
+)
+
+// Segmentize return modified Geography having no segment longer
+// that given maximum segment length.
+func Segmentize(geography *geo.Geography, segmentMaxLength float64) (*geo.Geography, error) {
+	spheroid := geographiclib.WGS84Spheroid
+	// Convert segmentMaxLength to s1.ChordAngle as further calculation
+	// is done considering segmentMaxLength as s1.ChordAngle.
+	segmentMaxLength = segmentMaxLength / spheroid.SphereRadius
+	geometry, err := geography.AsGeomT()
+	if err != nil {
+		return nil, err
+	}
+	segGeometry, err := segmentizeGeom(geometry, segmentMaxLength)
+	if err != nil {
+		return nil, err
+	}
+	return geo.NewGeographyFromGeom(segGeometry)
+}
+
+// segmentizeGeom returns a modified geom.T having no segment longer than
+// the given maximum segment length.
+func segmentizeGeom(geometry geom.T, segmentMaxLength float64) (geom.T, error) {
+	if segmentMaxLength <= 0 {
+		return nil, errors.Newf("maximum segment length must be positive")
```

Comment:
should this check be moved up to `Segmentize`?

Reply:
Sure,
This should be in `Segmentize` in the first place